src/main/java/com/xceptance/xlt/engine/util/UrlUtils.java

```diff
@@ -223,4 +225,37 @@ public static String getUrlEncodedParameters(final List<NameValuePair> parameter
         final List<org.apache.http.NameValuePair> httpClientPairs = NameValuePair.toHttpClient(parameters);
         return URLEncodedUtils.format(httpClientPairs, XltConstants.UTF8_ENCODING);
     }
+
+    /**
+     * Removes the user-info part from the given URL string.
+     * 
+     * @param url
+     *            the URL string
+     * @return given URL string without user-info part
+     */
+    public static String removeUserInfo(String url)
+    {
+        final URLInfo info = StringUtils.isNotBlank(url) ? parseUrlString(url) : null;
+        if (info != null)
+        {
+            final StringBuilder sb = new StringBuilder();
+            sb.append(info.getProtocol()).append("://").append(info.getHost());
+            if (info.getPort() > 0)
+            {
+                sb.append(':').append(info.getPort());
+            }
+            sb.append(info.getPath());
+            if (info.getQuery() != null)
+            {
+                sb.append('?').append(info.getQuery());
+            }
+            if (info.getFragment() != null)
+            {
+                sb.append('#').append(info.getFragment());
+            }
+            return sb.toString();
+        }
+
+        return null;
+    }
```

Comment:
Small performance optimization: We often have URL objects, not strings. What about providing a second version of `UrlUtils.removeUserInfo()` that takes a URL and builds the resulting user-less url string from the already parsed components in URL? This way we don't need to convert the URL to a string and afterwards parse the string again into its pieces.

Reply:
Done.